### PR TITLE
Add docs for integrating Memfault with Pinnacle 100 Firmware

### PR DESCRIPTION
**Pull request by chrisc11**

### CODE REVIEW

File: docs/mcu/pinnacle-100-guide.mdx [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+
+This tutorial assumes you have a working environment for building the
+[Pinnacle 100 Firmware](https://github.com/LairdCP/Pinnacle-100-Firmware-Manifest)
+(such as an an application based on the
+[AWS Out-of-Box (OOB) Demo](https://github.com/LairdCP/Pinnacle-100-Firmware/blob/main/docs/readme_ltem_aws.md#building-the-firmware)).
+
+## Create a Memfault Account
+
+The integration requires a Memfault cloud account to process data collected by
+the memfault-firmware-sdk. You can create one
```
Comment (fnune) [inline docs/mcu/pinnacle-100-guide.mdx:54]: Nit:

```suggestion
the Memfault Firmware SDK. You can create one
```

I believe this is how we call it elsewhere in the docs, unless it's a link.

File: docs/mcu/pinnacle-100-guide.mdx [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+CONFIG_MEMFAULT_NCS_PROJECT_KEY="YOUR_PROJECT_KEY"
+CONFIG_MEMFAULT_NCS_FW_VERSION_PREFIX="1.0.0+"
+CONFIG_MEMFAULT_NCS_FW_TYPE="pinnacle-fw"
+```
+
+## Publish data to the Memfault cloud
+
+### HTTPS
+
+The integration will automatically push data to Memfault over HTTPS when a LTE
```
Comment (fnune) [inline docs/mcu/pinnacle-100-guide.mdx:73]: Nit (unsure about this):

```suggestion
The integration will automatically push data to Memfault over HTTPS when an LTE
```

File: docs/mcu/pinnacle-100-guide.mdx [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+
+After a connection with AWS IoT is established, Memfault data is published to
+the following topic.
+
+```
+prod/<board>/<device_id>/memfault/<memfault_project_key>/chunk
+```
+
+Forwarding data to Memfault from AWS IoT is very easy! You just need to
+configure an AWS IoT Rule for the topic and connect it to a AWS Lambda function.
```
Comment (fnune) [inline docs/mcu/pinnacle-100-guide.mdx:88]: Nit (unsure about this):

```suggestion
configure an AWS IoT Rule for the topic and connect it to an AWS Lambda function.
```

Seems like there are two ways to say this? :smile: I'll stop commenting on these.

File: docs/mcu/pinnacle-100-guide.mdx [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+
+Next you will need to create a new
+[AWS IoT Rule](https://console.aws.amazon.com/iot/home#/rulehub):
+
+```
+SELECT encode(*, 'base64') AS data, topic() AS topic FROM 'prod/+/+/memfault/#'
+```
+
+Finally, as part of the setup you will need to connect the Lambda you just
+created to it so data is forwarded appropriately!
```
Comment (fnune) [inline docs/mcu/pinnacle-100-guide.mdx:301]: Having difficulty understanding what `it` refers to here.

**Review by fnune — APPROVED** [review]
Cool!

I mostly added some questionable nit-picks.

File: docs/mcu/pinnacle-100-guide.mdx [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+[AWS Lambda](https://console.aws.amazon.com/lambda/home#/functions) to forward
+data to Memfault
+
+```js
+const https = require("https");
+exports.https = https;
+
+exports.handler = (event, context, callback) => {
+  const data = Buffer.from(event.data, "base64");
+  const topicParams = event.topic.split("/");
```
[the remark this answers is not in the document]
Comment (chrisc11) [reply docs/mcu/pinnacle-100-guide.mdx:260]: 👍 Added a comment with the format of the MQTT topic

**Pull request merged**